### ISSUE TITLE
Child windows are not destroyed when the parent is destroyed

### ISSUE THREAD

**Issue by AndersMunch** [wxWidgets/Phoenix#630]
**Operating system**: Windows 10.
**wxPython version**: 4.0.0b2
**Stock or custom build**: No.
**Python version**: 2.7.14
**Stock or custom build**: No.

**Description of the problem**:
Child windows are automatically deleted when the parent is deleted, according to http://docs.wxwidgets.org/stable/overview_windowdeletion.html.  But that doesn't seem to apply to destroying frames in wxPython.

Run this, and close the parent window, then the parent's Destroy runs, but the child's doesn't.  

```python
import wx
class ChildFrame(wx.Frame):
    def __init__(self, parent):
        wx.Frame.__init__(self, parent, -1)
        wx.StaticText(self, -1, "child")
    def Destroy(self):
        print("Child Destroy")
        return wx.Frame.Destroy(self)
class ParentFrame(wx.Frame):
    def __init__(self, parent):
        wx.Frame.__init__(self, parent, -1)
        wx.StaticText(self, -1, "parent")
        ChildFrame(self).Show()
    def Destroy(self):
        print("Parent Destroy")
        return wx.Frame.Destroy(self)
app = wx.App()
fr = ParentFrame(None)
fr.Show(True)
app.MainLoop()
```

**Comment by RobinD42** [comment]
It is getting destroyed, which you can verify by binding the child's `EVT_DESTROY_WINDOW` event. The fact that the overridden `Destroy` method is not getting called is by design, as can be seen in this snippet of code from `DestroyChildren`:

```c++
        // note that we really want to delete it immediately so don't call the
        // possible overridden Destroy() version which might not delete the
        // child immediately resulting in problems with our (top level) child
        // outliving its parent
        child->wxWindowBase::Destroy();
```

So, in other words, `DestroyChildren` is bypassing your overridden `Destroy` method and instead is ensuring that the child will really be destroyed immediately.

**Comment by AndersMunch** [comment]
Thanks. I suspected the frame was actually being destroyed, but couldn't find any documentation to confirm it or explain why.

Overriding Destroy was actually my next try after failing to get EVT_WINDOW_DESTROY to work for me. The problem was that 
```python
self.Bind(wx.EVT_WINDOW_DESTROY, self.OnWindowDestroy)
```
also triggered the parent window event handler when the child window was destroyed, and the event object doesn't have any clear way of distinguising which frame is being destroyed. The wx._core.Frame returned by event.GetEventObject() does not compare equal to the frame.

Revisiting it, I think I now have the solution to how to run on-destroy code:
```python
self.Bind(wx.EVT_WINDOW_DESTROY, self.OnWindowDestroy, self)
```

Obvious in hindsight, but it took me a lot of experimentation to get to that point. Some more documentation would have been nice. I could write some myself if you'll give me wiki.wxpython.org access; I just created the wiki account AndersMunchFlonidan for this.
